app/graphql/types/query_type.rb

```diff
@@ -37,5 +40,46 @@ def news_item(id:)
     def tour(id:)
       Tour.find(id)
     end
+
+    # Provide contents from html files in `public/mobile-app/contents` through GraphQL query
+    #
+    # @param [String] name the file name
+    #
+    # @return [String] the contents of the file, if it exists - otherwise ""
+    def public_html_file(name:)
+      public_contents_folder = "#{Rails.root}/public/mobile-app/contents/"
+      file_type = "html"
+      file = "#{public_contents_folder}#{name}.#{file_type}"
+
+      if query_file?(file, public_contents_folder, file_type)
+        return { content: File.read(file).squish }
+      end
+
+      { content: "" }
+    end
+
+    private
+
+      # Provide a list of files within a folder that are allowed to query through GraphQL
+      #
+      # @param [String] path the path to lookup
+      # @param [String] file_type the file type to lookup
+      #
+      # @return [Array] list of queryable files
+      def query_files_whitelist(path, file_type)
+        queryable_files = File.join(path, "*.#{file_type}")
+        Dir.glob(queryable_files)
+      end
+
+      # Is a file allowed to query through GraphQL or not?
+      #
+      # @param [String] file the file to be queried
+      # @param [String] path the path to lookup
+      # @param [String] file_type the file type to lookup
+      #
+      # @return [Boolean] true, if file is existing and included in the whitelist - otherwise false
+      def query_file?(file, path, file_type)
+        File.exist?(file) && query_files_whitelist(path, file_type).include?(file)
+      end
```

Comment:
`public_contents_folder` wird oben dynamisch definiert und dann in mehrere Methoden weiterschleift.
Eigentlich soll und darf sich `public_contents_folder` aber gar nicht ändern sondern ist fix gesetzt auf  
"#{Rails.root}/public/mobile-app/contents" oder?

wenn ja, dann würde ich oben eher eine Konstante Definieren auf die die Methoden unten zugreifen. und mir den Übergabeparameter sparen

Reply:
ja guter gedanke. aber ich spinne es mal weiter.
würde es nur um die html contents gehen, wäre das möglich.
ich habe aber bewusst `query_files_whitelist` und `query_file?` allgemein gehalten. so sind diese wiederverwendbar für andere ordner und andere dateitypen.

für #41 benötigen wir selbige logiken für `.json` dateien in einem anderen ordner, vermutlich `#{Rails.root}/public/mobile-app/configs`.